californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -679,17 +679,20 @@ private void reject(final RawData raw, final MessageFormatException cause) {
 			rst.setToken(new byte[0]);
 			rst.setDestination(raw.getAddress());
 			rst.setDestinationPort(raw.getPort());
-			for (MessageInterceptor interceptor : interceptors) {
-				interceptor.sendEmptyMessage(rst);
-			}
-			connector.send(serializer.serializeEmptyMessage(rst));
+
+			reject(rst);
```

Comment:
I don't think doing this in the Matcher will work. This particular code path bypasses Stack because it tries to manually send a RST for a badly formed message.

One option that I like a little more now, is to just send RST through the stack. I think that should have no side effect, because non of the layers do anything stateful with RST message (and in fact, may have positive side effect by clearing any state in the matcher). 

And that will solve the problem of infinite ping pong, because it TcpStack will correctly ignore RST message. Thoughts?